### PR TITLE
Move FIS precondition format check to installation constructor

### PR DESCRIPTION
**Pull request by ChaoqunCHEN**

### CODE REVIEW

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java [hunk 1]
```diff
@@ -142,17 +142,17 @@ public Thread newThread(Runnable r) {
    * FirebaseOptions#getApplicationId()} , and ({@link FirebaseOptions#getProjectId()} are not null
    * or empty.
    */
-  private void preConditionChecks() {
-    Preconditions.checkNotEmpty(getApplicationId());
-    Preconditions.checkNotEmpty(getProjectIdentifier());
-    Preconditions.checkNotEmpty(getApiKey());
```
Comment (ankitaj224) [inline firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:148]: Did you remove these methods?

getApplicationId, getProjectIdentifier, getApiKey

Also, for getProjectIdentifier we wanted to either use project id or sender id.

Comment (ChaoqunCHEN) [reply firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:148]: - No, I didn't remove those methods. I just made preConditionChecks to be static and move it to constructor from getId() and getToken().

- getProjectIdentifier was changed to take only projectId on last PR as requested by Rayo.

Comment (ankitaj224) [reply firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:148]: nit: If those methods arent used remove them.

Comment (ChaoqunCHEN) [reply firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java:148]: Thanks, those methods are used somewhere else. So I keep them.

**Pull request merged**